src/utils/number_utils.ts

```diff
@@ -7,6 +9,13 @@ const FLOAT_BYTES = new Uint8Array(FLOAT.buffer, 0, 8);
  * @internal
  */
 export const NumberUtils = {
+  getSize(source: Uint8Array, offset: number): number {
```

Comment:
Does this belong in `NumberUtils`?  The utils mainly abstract reading/writing to and from buffers, but no other function here has any logic specific to BSON.  Maybe we could put this in a floating function instead?

also - maybe `getBSONSizeAtOffset`?  `getSize(uint8array)` sort of implies you're getting the size of the array, not the size of a BSON element

Reply:
I don't feel strongly, this does produce a BSON specific error but I saw it as a `size_t` kind of parser, so still falling under the category of "number"

Anyway, I did orginally have it outside NumberUtils, happy to move it back out.